GeneratorInterface/Core/interface/HadronizerFilter.h

```diff
@@ -269,6 +282,25 @@ namespace edm
       finalEvent->weights()[0] *= multihadweight;
     }
     
+    if (secondaryUnweighting_) {
+      double weight = finalGenEventInfo->weights()[0];
+      double absweight = std::abs(weight);
+      if (absweight<maxAbsWeight_) {
+        double randomflat = randomEngineSentry.randomEngine()->flat();
```

Comment:
To be clear, this same randomEngineSentry is also being used by pythia itself to generate its own random numbers for the same event.  (But indeed I do not see any problem with that.)